code/parachain/frame/crowdloan-rewards/src/lib.rs

```diff
@@ -307,12 +315,35 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
+		/// Initialize the Crowdloan at a given block.
+		///
+		/// If the Crowdloan is over funded by more than the `OverFundedThreshold`, the `OverFunded`
+		/// event will be emitted with the excess amount.
+		///
+		/// # Errors
+		/// * `AlreadyInitialized` - The Crowdloan has already been scheduled to start at some block
+		/// * `BackToTheFuture` - The given block, `at`, is before the current block
+		/// * `RewardsNotFunded` - The Crowdloan has not been funded with the minimum amount of
+		///   funds to provide the total rewards
 		pub(crate) fn do_initialize(at: MomentOf<T>) -> DispatchResult {
 			ensure!(!VestingTimeStart::<T>::exists(), Error::<T>::AlreadyInitialized);
+
 			let now = T::Time::now();
 			ensure!(at >= now, Error::<T>::BackToTheFuture);
+
+			let available_funds = T::RewardAsset::balance(&Self::account_id());
+			let total_rewards = TotalRewards::<T>::get();
+			let excess_funds = available_funds
+				.checked_sub(&total_rewards)
+				.ok_or(Error::<T>::RewardsNotFunded)?;
+
+			if excess_funds > T::OverFundedThreshold::get().mul_floor(total_rewards) {
+				Self::deposit_event(Event::OverFunded { excess_funds })
```

Comment:
so it will spam continuously in case of issue. seem better place is logs.

Reply:
This isn't a hook, it only emits when the extrinsic is called

Reply:
oh. fun. then why not error. nwmd

Reply:
if that is not background, why not error? just qurious - may be that is some patter to follow

Reply:
Because it's not an error to have it over funded, this event is just a notification really

Reply:
okey. i have arguments it should not be here than. but whatever:)